backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/BPartnerPrintFormatRepository.java

```diff
@@ -87,21 +126,38 @@ public BPPrintFormat save(@NonNull final BPPrintFormat bpPrintFormat)
 
 	private I_C_BP_PrintFormat createOrUpdateRecord(@NonNull final BPPrintFormat bpPrintFormat)
 	{
-		final I_C_BP_PrintFormat bpPrintFormatRcord;
+		final I_C_BP_PrintFormat bpPrintFormatRecord;
 		if (bpPrintFormat.getBpPrintFormatId() > 0)
 		{
-			bpPrintFormatRcord = load(bpPrintFormat.getBpPrintFormatId(), I_C_BP_PrintFormat.class);
+			bpPrintFormatRecord = load(bpPrintFormat.getBpPrintFormatId(), I_C_BP_PrintFormat.class);
 		}
 		else
 		{
-			bpPrintFormatRcord = newInstance(I_C_BP_PrintFormat.class);
+			bpPrintFormatRecord = newInstance(I_C_BP_PrintFormat.class);
 		}
 
-		bpPrintFormatRcord.setC_BPartner_ID(bpPrintFormat.getBpartnerId().getRepoId());
-		bpPrintFormatRcord.setC_DocType_ID(bpPrintFormat.getDocTypeId());
-		bpPrintFormatRcord.setAD_Table_ID(bpPrintFormat.getAdTableId());
-		bpPrintFormatRcord.setAD_PrintFormat_ID(bpPrintFormat.getPrintFormatId());
+		bpPrintFormatRecord.setC_BPartner_ID(bpPrintFormat.getBpartnerId().getRepoId());
+		if(bpPrintFormat.getDocTypeId() != null)
```

Comment:
when saving back to DB, always set each field!
e.g. do
```java
bpPrintFormatRecord.setC_DocType_ID(DocTypeId.toRepoId(bpPrintFormat.getDocTypeId())
```
instead of checking and setting it only when it's not null.

Reply:
ok thx, fixed